docs/tasks/access-kubernetes-api/migrate-third-party-resource.md

```diff
@@ -0,0 +1,163 @@
+    you with TPR.
+
+    For example, if your ThirdPartyResource looks like this:
+
+    ```yaml
+    apiVersion: extensions/v1beta1
+    kind: ThirdPartyResource
+    metadata:
+      name: cron-tab.stable.example.com
+    description: "A specification of a Pod to run on a cron style schedule"
+    versions:
+    - name: v1
+    ```
+
+    A matching CustomResourceDefinition could look like this:
+
+    ```yaml
+    apiVersion: apiextensions.k8s.io/v1beta1
+    kind: CustomResourceDefinition
+    metadata:
+      name: crontabs.stable.example.com
+    spec:
+      scope: Namespaced
+      group: stable.example.com
+      version: v1
+      names:
+        kind: CronTab
+        plural: crontabs
+        singular: crontab
+    ```
+
+1.  **Install the CustomResourceDefinition**
+
+    While the source TPR is still active, install the matching CRD with `kubectl create`.
+    Existing TPR data will remain accessible because TPRs take precedence over CRDs when both try
+    to serve the same resource.
+
+    After you create the CRD, make sure the *Established* condition goes to True.
+    You can check it with a command like this:
+
+    ```shell
+    kubectl get crd -o 'custom-columns=NAME:{.metadata.name},ESTABLISHED:{.status.conditions[?(@.type=="Established")].status}'
+    ```
+
+    The output should look like this:
+
+    ```console
+    NAME                          ESTABLISHED
+    crontabs.stable.example.com   True
+    ```
+
+1.  **Stop all clients that use the TPR**
+
+    The API server will attempt to prevent mutation of TPR data for the resource while it
+    copies objects to the CR, but it can't guarantee consistency in all cases, such as with
+    [multiple masters](/docs/admin/high-availability/).
+    Stopping clients, such as TPR-based custom controllers, helps to avoid inconsistencies in
+    the copied data.
+
+    In addition, clients that watch TPR data will not receive any more events once the migration
+    begins.
+    You must restart them after the migration completes so they start watching CRD data instead.
+
+1.  **Back up TPR data**
+
+    In case the data migration fails, save a copy of existing data for the resource:
+
+    ```shell
+    kubectl get crontabs --all-namespaces -o yaml > crontabs.yaml
+    ```
+
+    You should also save a copy of the TPR definition if you don't have one already:
+
+    ```shell
+    kubectl get thirdpartyresource cron-tab.stable.example.com -o yaml --export > tpr.yaml
+    ```
+
+1.  **Delete the TPR definition**
+
+    Normally, when you delete a TPR definition, the API server tries to clean up any objects stored
+    in that resource.
+    Since you created a matching CRD, the server will copy objects to the CRD instead of deleting
+    them.
+
+    ```shell
+    kubectl delete thirdpartyresource cron-tab.stable.example.com
+    ```
+
+1.  **Verify the new CRD data**
+
+    It can take up to 10 seconds for the TPR controller to notice that you deleted the TPR definition
```

Comment:
[RW] It can take up to 10 seconds for the TPR controller to notice that you delete the TPR definition and to initiate the migration. The data remains accessible during this time.

Once the migration completes, the resource begins serving through the CRD.

avoids future tense and adds parallel construction